### PR TITLE
PT-515 Add capacity to display YLP version coming from env

### PR DESCRIPTION
**Pull request by BigFab**
Modifications for Health-check before modifications on YourLoops

### CODE REVIEW

File: service.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ var request = require('request-promise');
 const bunyan = require("bunyan");
 
 class HealthCheckService {
-    constructor(services) {
+    constructor(ylpVersion, services) {
```
Comment (ydrisR) [inline service.js]: You will need to update the tests to use this new signature, OR to switch both variables

**Review by ydrisR — CHANGES_REQUESTED** [review]
Tests are required ;)
* reflect the new signature of the constructor
* add a test in test_service to verify the version of yourloops is returned by the service
* add a test in test_env to verify the new env variable is retrieved as expected by the program

**Comment by BigFab** [comment]
Existing tests have been fixed and new ones have been added

File: env.js [hunk 2]
```diff
@@ -20,9 +20,12 @@ module.exports = (function() {
   var env = {};
   //port on which to expose the service
   env.servicePort = process.env.SERVICE_PORT || 8080;
+  // YourLoops version the health-check is running on
+  env.ylpVersion = process.env.YLP_VERSION;
```
Comment (ydrisR) [inline env.js:24]:
```suggestion
  env.ylpVersion = process.env.YLP_VERSION || "unknown";
```

File: service.js [hunk 3]
```diff
@@ -97,14 +103,19 @@ class HealthCheckService {
                 if (totalOfCheckedServices == nbOfServices) {
                     if (isOK) {
                         this.logger.info("all services are up and running!");
```
Comment (ydrisR) [inline service.js:115]:
```suggestion
                        this.logger.info("all services are up and running!");
                        ylp['status']="OK";
```

File: service.js [hunk 4]
```diff
@@ -97,14 +103,19 @@ class HealthCheckService {
...
-                        this.logger.debug(globalResult);
-                        res.status(200);
-                        res.send(globalResult);
                     } else {
                         this.logger.info("at least one service is down");
-                        res.status(503);
-                        res.send(globalResult);
+                        // at least one service down and we return 503
+                        responseStatus = 503;
```
Comment (ydrisR) [inline service.js]:
```suggestion
                        responseStatus = 503;
                        ylp['status'] = "NOK";
```

File: service.js [hunk 5]
```diff
@@ -97,14 +103,19 @@ class HealthCheckService {
...
                     }
+                    // Overall status of YLP depends on all services status
+                    ylp['status'] = responseStatus;
```
Comment (ydrisR) [inline service.js]:
```suggestion
```

File: test/test_service.js [hunk 6]
```diff
@@ -26,24 +27,31 @@ describe('Global health check service', () => {
...
         });
-        it('all 3 services should report a detail = "OK"', () => {
-            statusResults[0].details.should.eql("OK");
+        it('all 3 services (excluding "yourloops") should report a detail = "OK"', () => {
             statusResults[1].details.should.eql("OK");
             statusResults[2].details.should.eql("OK");
+            statusResults[3].details.should.eql("OK");
+        });
+        it('"yourloops" item should report a status = 200', () => {
+            statusResults[0].status.should.eql(200);
```
Comment (ydrisR) [inline test/test_service.js]:
```suggestion
            statusResults[0].status.should.eql("OK");
```

File: test/test_service.js [hunk 7]
```diff
@@ -55,20 +63,26 @@ describe('Global health check service', () => {
...
             });
         });
         it('should return an error for shoreline', () => {           
             let authStatus = statusResults.filter(status => status.service == "shoreline");
             authStatus[0].status.should.eql("NOK");
             authStatus[0].error.should.eql("500 - \"service is down\"");
-        })
+        });
+        it('"yourloops" item should report a status = 503', () => {
+            statusResults[0].status.should.eql(503);
```
Comment (ydrisR) [inline test/test_service.js]:
```suggestion
            statusResults[0].status.should.eql("NOK");
```

**Review by ydrisR — CHANGES_REQUESTED** [review]
Nice improvement :)
A couple of changes (to discuss if you disagree):
1) Either set a default value for env.ylpVersion or throw an error at startup time if the environment variable is null or empty. 
2) Why not return OK or NOK as a status for the yourloops entry instead of 200/503? That would be nice to have the same convention for all services, and even if we know what 200/503 means, OK/NOK is a bit clearer I think.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,9 +1,14 @@
-# Yourloops Health Check agregator service
+# Yourloops Health Check aggregator service
 
-Health-Check is a simple service which allows you to monitor several web services and provide an agregated status.
-The service exposes this agregated status on route /status
+Health-Check is a simple service which allows you to monitor several web services and provide an aggregated status.
+The service exposes this aggregated status on route /status
 This end point is typically what would be provided to a load balancer to check the health of the entire stack.
 
+## [Unreleased]
```
Comment (ydrisR) [inline CHANGELOG.md]:
```suggestion
## 1.1.0 - 2019-07-31
```

**Review by ydrisR — CHANGES_REQUESTED** [review]

File: test/test_service.js [hunk 9]
```diff
@@ -20,35 +20,35 @@ describe('Global health check service', () => {
     before((done) => {
         //prepare services mocking
         services.get('/auth/status').reply(200, "OK")
-                .get('/access/status').reply(200,"\"OK\"")
-                .get('/data/status').reply(200,"OK");
+                .get('/access/status').reply(200, "OK")
```
Comment (ydrisR) [inline test/test_service.js]: hmm hmm, the double quotes had a purpose ;)

**Review by ydrisR — CHANGES_REQUESTED** [review]
Almost perfect ;)

**Review by ydrisR — APPROVED** [review]
🚀 🎆

**Pull request merged**